speedy-recovery-client/src/service/FhirDataFilteringService.js

```diff
@@ -0,0 +1,73 @@
+const filterPatientResource = resource => {
+  // resourceType, surname, careprovider
+  if (
+    resource &&
+    resource.name &&
+    resource.name[0] &&
+    resource.name[0].family
+  ) {
+    const requiredFields = [resource.resourceType, resource.name[0].family];
```

Comment:
Now these methods are certainly more resilient. However, after the `if` conditions validated as truthy, you already know what you want to know. No need to check the required fields again right? I guess you can just return true then, and false otherwise.

Reply:
will fix this shortly